apps/src/code-studio/pd/foorm/FoormLibraryEditorManager.jsx

```diff
@@ -23,81 +25,125 @@ const styles = {
-      formPreviewQuestions: null,
       showCodeMirror: false,
       hasLoadError: false
     };
 
-    this.props.resetAvailableForms(this.props.formNamesAndVersions);
-  }
-
-  componentDidUpdate(prevProps) {
-    if (prevProps.formNamesAndVersions !== this.props.formNamesAndVersions) {
-      this.props.resetAvailableForms(this.props.formNamesAndVersions);
-    }
+    this.props.resetAvailableLibraries(this.props.libraryNamesAndVersions);
   }
 
+  // could probably dedupe this method
   getFormattedConfigurationDropdownOptions() {
-    return this.props.availableForms.map((formNameAndVersion, i) => {
-      const formName = formNameAndVersion['name'];
-      const formVersion = formNameAndVersion['version'];
-      const formId = formNameAndVersion['id'];
-      return (
-        <MenuItem
-          key={i}
-          eventKey={i}
-          onClick={() => this.loadConfiguration(formId)}
-        >
-          {`${formName}, version ${formVersion}`}
-        </MenuItem>
+    return this.props.availableLibraries.map((libraryNameAndVersion, i) => {
+      const libraryName = libraryNameAndVersion['name'];
+      const libraryVersion = libraryNameAndVersion['version'];
+
+      return this.getMenuItem(
+        () => this.loadLibraryQuestions(libraryNameAndVersion),
+        `${libraryName}, version ${libraryVersion}`,
+        i
       );
     });
   }
 
-  loadConfiguration(formId) {
+  getFormattedLibraryQuestionDropdownOptions() {
+    return this.props.availableLibraryQuestionsForCurrentLibrary.map(
+      (libraryQuestionAndType, i) => {
+        const libraryQuestionName = libraryQuestionAndType['name'];
+        const libraryQuestionType = libraryQuestionAndType['type'];
+        const libraryQuestionId = libraryQuestionAndType['id'];
+
+        return this.getMenuItem(
+          () => this.loadLibraryQuestionContent(libraryQuestionId),
+          `${libraryQuestionName} (${libraryQuestionType})`,
+          i
+        );
+      }
+    );
+  }
+
+  getMenuItem = (clickHandler, textToDisplay, key) => {
+    return (
+      <MenuItem key={key} eventKey={key} onClick={clickHandler}>
+        {textToDisplay}
+      </MenuItem>
+    );
+  };
+
+  loadLibraryQuestions(selectedLibraryNameAndVersion) {
+    let libraryId = selectedLibraryNameAndVersion['id'];
+
     this.props.setLastSaved(null);
     this.props.setSaveError(null);
+    this.props.setLibraryData(selectedLibraryNameAndVersion);
+    this.updateLibraryQuestionData({
+      question: {},
+      name: null,
+      id: null
+    });
+
     $.ajax({
-      url: `/api/v1/pd/foorm/forms/${formId}`,
+      url: `/foorm/libraries/${libraryId}/question_names`,
       type: 'get'
     })
       .done(result => {
-        this.updateFormData(result);
+        this.props.setHasError(false);
+        this.props.resetAvailableLibraryQuestions(result);
+        this.setState({
+          hasLoadError: false
+        });
+      })
+      .fail(() => {
+        this.setState({
```

Comment:
Do you need to also set `this.props.setHasError(true)` here? What's the difference?

Reply:
Ok so I think I made a mistake including this `setHasError(false)` in the first place, errors are confusing here. There are 3? maybe more? error states at play in this editor:

hasError (redux): keeps track of whether the [JSON in the editor](https://github.com/code-dot-org/code-dot-org/blob/6ea535f818f8047aa2b9fac426c2055508c7d582/apps/src/sites/studio/pages/foorm/libraries/editor.js#L50) (eg, missing quotes, etc.) is valid.
saveError (redux): tracks any error produced when attempting to save an edited/new library question to the database.
hasLoadError (state local to the `FoormLibraryEditorManager` component): tracks if an error was produced when trying to fetch a library question or library from the database.

Reply:
may i suggest renaming `hasError` to `jsonError` or `editingError` or something like that to make it sound less like a catch-all?